runtime/runtime/src/lib.rs

```diff
@@ -525,6 +539,10 @@ impl Runtime {
         if receipt.predecessor_id == system_account() {
             result.gas_burnt = 0;
             result.gas_used = 0;
+            if result.result.is_err() {
```

Comment:
If the refund fails, instead of just burning tokens, we report the total number of tokens burnt in the ApplyResult. It can be used by validators to distribute it. Added a comment